h/api/nipsa/search.py

```diff
@@ -0,0 +1,61 @@
+def nipsa_filter(userid=None):
+    """Return an Elasticsearch filter for filtering out NIPSA'd annotations.
+
+    The returned filter is suitable for inserting into an Es query dict.
+    For example:
+
+        query = {
+            "query": {
+                "filtered": {
+                    "filter": nipsa_filter(),
+                    "query": {...}
+                }
+            }
+        }
+
+    :param userid: The ID of a user whose annotations should not be filtered.
+        The returned filtered query won't filter out this user's annotations,
+        even if the annotations have the NIPSA flag.
+    :type userid: unicode
+
+    """
+    # If any one of these "should" clauses is true then the annotation will
+    # get through the filter.
+    should_clauses = [{"not": {"term": {"nipsa": True}}}]
+
+    if userid:
```

Comment:
I think what this comes down to is what we think it should do if the user code passes in an empty string for the userid: current behaviour is not to add any userid filter, the Es query will work as if there's no authorized user. If I changed it to `is None` it would add a `{"term": {"user": ""}}` filter which I'm guessing will cause a traceback from Es. Or we could explicitly check for the empty string and raise our own exception here. I think the current implementation is as defensible as any of the other options, isn't it?